man/systemd.netdev.xml

```diff
@@ -983,6 +986,100 @@
     as the <literal>[Tun]</literal> section.</para>
   </refsect1>
 
+  <refsect1>
+    <title>[WireGuard] Section Options</title>
+
+    <para>The <literal>[WireGuard]</literal> section accepts the following
+    key:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>PrivateKey=</varname></term>
+        <listitem>
+          <para>The private key for the interface. It can be generated
+            using the <command>wg genkey</command> command. This option is
+            mandatory to use wireguard.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>ListenPort</varname></term>
+        <listitem>
+          <para>Sets UDP port for listening. Takes either value between 1 and 65535
+            or <literal>auto</literal>. If <literal>auto</literal> is specified,
+            the port is automatically generated based on interface name.
+            Defaults to <literal>auto</literal>.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>FwMark</varname></term>
+        <listitem>
+          <para>Sets a firewall mark on outgoing wireguard packets from this interface.</para>
+        </listitem>
+      </varlistentry>
+    </variablelist>
+  </refsect1>
+
+  <refsect1>
+    <title>[WireGuardPeer] Section Options</title>
+
+    <para>The <literal>[WireGuardPeer]</literal> section accepts the following
+    key:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>PublicKey=</varname></term>
+        <listitem>
+          <para>Sets a base64 public key calculated by <command>wg pubkey</command> from a
+              private key, and usually transmitted out of band to the author
+              of the configuration file. This option is mandatory for this section.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>PresharedKey=</varname></term>
+        <listitem>
+          <para>Optional preshared key for the interface. It can be generated
+            by the <command>wg genpsk</command> command. This option adds an
+            additional layer of symmetric-key cryptography to be mixed into the
+            already existing public-key cryptography, for post-quantum
+            resistance.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>AllowedIPs=</varname></term>
+        <listitem>
+          <para>Sets a comma-separated list of IP (v4 or v6) addresses with CIDR masks
+            from which this peer is allowed to send incoming traffic and to
+            which outgoing traffic for this peer is directed. The catch-all
+            0.0.0.0/0 may be specified for matching all IPv4 addresses, and
+            ::/0 may be specified for matching all IPv6 addresses. </para>
```

Comment:
to make this similar to IPAddressAllow= this should also understand the aliases `any`... In fact I think we can share some code here, i.e. split out some parts of config_parse_ip_address_access() into a new function `int in_addr_prefix_alias_from_string(const char *string, struct in_addr *ret_ipv4, unsigned char *ret_ipv4_prefixlen, struct in6_addr *ret_ipv6, unsigned char *ret_ipv6_prefixlen);` that translates "any", "link-local", "multicast", "localhost" into a pair of addresses with corresponding masks.

Reply:
Except it's not the same thing to IPAddressAllow, and trying to map the concepts to each other will cause more confusion than it causes.